src/clr-angular/utils/drag-and-drop/droppable/droppable.ts

```diff
@@ -53,12 +53,42 @@ export class ClrDroppable<T> implements OnInit, OnDestroy {
     }
 
     @Input("clrDropTolerance")
-    set dropTolerance(value: number|ClrDropTolerance) {
+    set dropTolerance(value: number|string|ClrDropTolerance) {
         if (typeof value === "number") {
             this._dropTolerance.top = value;
             this._dropTolerance.right = value;
             this._dropTolerance.bottom = value;
             this._dropTolerance.left = value;
+        } else if (typeof value === "string") {
+            const toleranceValues = value.trim().split(/\s+/).map((tolerance) => parseInt(tolerance, 10));
+            switch (toleranceValues.length) {
+                case 1:
```

Comment:
Just showing off 😎:
```typescript
const nbValues = toleranceValues.length;
['top', 'right', 'bottom', 'left'].forEach((dir, i) => {
	this._dropTolerance[dir] = toleranceValues[Math.max(i%nbValues, i%2%nbValues)];
});
```

That can replace your entire switch.

Reply:
Update as per our discussion on Slack.